client/header/activity-panel/index.js

```diff
@@ -70,30 +66,35 @@ export class ActivityPanel extends Component {
 		};
 	}
 
-	togglePanel( tabName ) {
-		const { isPanelOpen, currentTab } = this.state;
-
-		// If a panel is being opened, or if an existing panel is already open and a different one is being opened, record a track.
-		if ( ! isPanelOpen || tabName !== currentTab ) {
-			recordEvent( 'activity_panel_open', { tab: tabName } );
-		}
-
+	togglePanel( { name: tabName }, isTabOpen ) {
 		this.setState( ( state ) => {
-			if ( tabName === state.currentTab || state.currentTab === '' ) {
-				return {
-					isPanelOpen: ! state.isPanelOpen,
-					currentTab: tabName,
-					mobileOpen: ! state.isPanelOpen,
-				};
-			}
-			return { currentTab: tabName, isPanelSwitching: true };
+			const isPanelSwitching =
+				tabName !== state.currentTab &&
+				state.currentTab !== '' &&
+				isTabOpen &&
+				state.isPanelOpen;
+
+			return {
+				isPanelOpen: isTabOpen,
+				mobileOpen: isTabOpen,
+				currentTab: tabName,
+				isPanelSwitching,
+			};
 		} );
 	}
 
+	closePanel() {
+		this.setState( ( state ) => ( {
```

Comment:
one thing that was happening a lot before was that ActivityPanel was setting state to be a completely different shaped object in various scenarios. This lead to a lot of confusion about what state was at any given time. I've changed this to always keep state the same shape.